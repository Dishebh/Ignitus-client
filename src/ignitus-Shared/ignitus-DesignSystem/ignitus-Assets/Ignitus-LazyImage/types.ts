export interface LazyImageTypes {
  alt: string;
  src: string;
  height?: string;
  width?: string;
  width0?: string;
  width1?: string;
  width2?: string;
  width3?: string;
  width4?: string;
  width5?: string;
  height1?: string;
  height2?: string;
  height5?: string;
  padding?: string;
  maxWidth?: string;
  borderRadius?: string;
}
