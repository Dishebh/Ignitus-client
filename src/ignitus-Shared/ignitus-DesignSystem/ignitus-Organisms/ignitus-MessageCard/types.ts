export type Props = {
  name: string;
  position: string;
  numberMessages: string;
  dateSent: string;
  avatar: string;
  online: boolean;
};
