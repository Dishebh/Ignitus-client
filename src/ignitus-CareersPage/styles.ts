import styled from '@emotion/styled';
import { minimumWidthQuery, IgnitusBlue } from '../ignitus-Shared';
import { RowTypes } from './types';
import {
  Container as C,
  flexibleColDiv,
} from '../ignitus-Shared/ignitus-DesignSystem/shared';

export const TextContainer = styled.div`
  text-align: center;
`;

export const TopSection = styled.div`
  background: ${IgnitusBlue};
  padding: 8rem 4rem 4rem;
  text-align: center;
`;

export const ParagraphWrapper = styled.div`
  text-align: ${(props: { alignment?: string }) => props.alignment};
  padding: 0.5rem 0;
`;

export const PerksParagraphWrapper = styled(ParagraphWrapper)`
  width: 15rem;
`;

export const HeadingWrapper = styled.div`
  padding: 1rem 0;
  text-align: ${(props: { alignment?: string }) => props.alignment};
`;

export const Container = styled(C)`
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
`;

export const Section = styled(flexibleColDiv)`
  margin: 2rem 0;
`;

export const LeftRow = styled.div<RowTypes>`
  flex: ${props => props.flex};
  display: flex;
  align-items: center;
  justify-content: ${props => props.alignment};
  ${minimumWidthQuery[0]} {
    margin: 1rem 0;
  }
  ${minimumWidthQuery[3]} {
    margin: 0;
  }
`;
export const RightRow = styled.div<RowTypes>`
  flex: ${props => props.flex};
  text-align: left;
  ${minimumWidthQuery[0]} {
    margin: 1rem 0;
  }
  ${minimumWidthQuery[3]} {
    margin: 0;
  }
`;

export const PerksImageWrapper = styled.div`
  padding: 0.5rem;
  text-align: left;
`;

export const OpportunityWrapper = styled.div`
  margin: 2rem 0;
`;

export const OpportunitySection = styled.div`
  width: 100%;
`;

export const PerksWrapper = styled.div`
  padding: 1rem;
  display: flex;
  flex-direction: row;
  align-items: center;
  flex-wrap: wrap;

  ${minimumWidthQuery[0]} {
    flex: 1 1 100%;
  }
  ${minimumWidthQuery[1]} {
    flex: 1 1 100%;
  }
  ${minimumWidthQuery[2]} {
    flex: 1 1 50%;
  }
  ${minimumWidthQuery[3]} {
    flex: 1 1 30%;
  }
`;

export const Row = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-around;
  flex-direction: ${(props: { direction?: string }) => props.direction};

  ${minimumWidthQuery[0]} {
    margin: 2rem 0;
    flex-direction: column;
  }
  ${minimumWidthQuery[3]} {
    flex-direction: ${(props: { direction?: string }) => props.direction};
    margin: 0;
  }
`;

export const PerksRow = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-around;

  ${minimumWidthQuery[0]} {
    margin: 2rem 0;
    flex-direction: column;
  }
  ${minimumWidthQuery[3]} {
    flex-direction: row;
    margin: 0;
  }
`;
