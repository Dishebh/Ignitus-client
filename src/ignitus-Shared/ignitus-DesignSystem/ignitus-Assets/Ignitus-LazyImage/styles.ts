import styled from '@emotion/styled';

const placeHolder =
  'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkqAcAAIUAgUW0RjgAAAAASUVORK5CYII=';

export const Image = styled.img`
  display: block;
  height: 100px;
  width: 100px;
  // Add a smooth animation on loading
  @keyframes loaded {
    0% {
      opacity: 0.1;
    }
    100% {
      opacity: 1;
    }
  }
  // I use utilitary classes instead of props to avoid style regenerating
  &.loaded:not(.has-error) {
    animation: loaded 300ms ease-in-out;
  }
  &.has-error {
    // fallback to placeholder image on error
    content: url(${placeHolder});
  }
`;
