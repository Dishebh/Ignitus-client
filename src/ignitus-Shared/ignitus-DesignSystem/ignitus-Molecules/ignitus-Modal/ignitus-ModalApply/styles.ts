import styled from '@emotion/styled';
import {
  Paragraph as P,
  Normal,
  MD,
  White,
  boxShadowColor,
  SecondaryColor,
} from '../../../..';

import { flexibleColDiv, flexibleRowDiv } from '../../../shared';
import { RoundedButton } from '../../../ignitus-Atoms/buttons';
import { DefaultTag1 } from '../../../ignitus-Atoms/ignitus-defaultTag/Components/defaultTag';

export const ModalApply = styled.div`
  background-color: ${White};
  border-radius: 1rem;
  display: inline-flex;
  flex-direction: column;
  box-shadow: 0 2px 4px 0 ${boxShadowColor};
  padding: 2rem 4rem;
`;

export const TopSection = styled(flexibleColDiv)`
  align-items: flex-start;
`;

export const MiddleSection = styled.div`
  display: flex;
  flex-direction: row;
  margin-top: 1.5rem;
`;

export const LeftSection = styled(flexibleColDiv)`
  align-items: flex-start;
`;

export const RightSection = styled(flexibleColDiv)`
  align-items: flex-start;
  margin-left: 2rem;
`;

export const BottomSection = styled(flexibleRowDiv)`
  justify-content: flex-end;
  margin-top: 1.5rem;
`;

export const Paragraph = styled(P)`
  font-weight: ${Normal};
  font-size: ${MD};
`;

export const StyledRoundedButton = styled(RoundedButton)`
  background: none;
  border: 2px solid ${SecondaryColor};
  color: ${SecondaryColor};
`;

export const StyledDefaultTag1 = styled(DefaultTag1)`
  padding: 0.7rem 1rem;
`;
