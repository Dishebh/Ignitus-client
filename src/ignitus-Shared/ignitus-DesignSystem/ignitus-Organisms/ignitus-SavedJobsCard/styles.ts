import styled from '@emotion/styled';
import * as C from '../../ignitus-Atoms/colors';
import * as T from '../../ignitus-Atoms/typography';
import * as F from '../../ignitus-Atoms/fonts';
import { flexibleRowDiv, flexibleColDiv } from '../../shared';
import { Icon as I } from '../../../ignitus-Utilities/Components/icon';

export const HeaderContainer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 1.75rem;
`;

export const SavedJobContainer = styled(flexibleRowDiv)`
  border-radius: 1rem;
  box-shadow: 0 3px 6px 0 ${C.boxShadowColor};
  background-color: ${C.White};
  padding-left: 1.5rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  width: 100%;
  margin-bottom: 1.25rem;
  flex-direction: row;
  justify-content: flex-start;
  display: flex;
  position: relative;
`;

export const ImageContainer = styled.div`
  position: relative;
  img {
    width: 4rem;
    max-width: 100%;
    max-height: 100%;
  }
`;

export const DetailContainer = styled(flexibleColDiv)`
  height: 100%;
  margin-left: 0.7rem;
  align-items: flex-start;
`;

export const JobTitle = styled(T.Paragraph)`
  font-size: ${F.MD};
  font-family: ${F.PrimaryFamily};
  font-weight: ${F.Bold};
  color: ${C.SecondaryColor};
`;

export const JobDesc = styled(T.Paragraph)`
  font-size: ${F.XS};
  font-family: ${F.PrimaryFamily};
  font-wieght: ${F.Light};
  color: ${C.IgnitusBlue};
  margin-top: 2px;
  margin-bottom: 2px;
`;

export const JobUpdate = styled(T.Paragraph)`
  font-size: ${F.XS};
  font-family: ${F.PrimaryFamily};
  font-wieght: ${F.Light};
  color: ${C.GreySecondaryText};
`;

export const Icon = styled(I)``;

export const IconContainer = styled.div`
  &:hover {
    cursor: pointer;
  }
`;

export const DropdownContainer = styled(flexibleColDiv)`
  border-radius: 1rem 0 1rem 1rem;
  width: 10.3rem;
  box-shadow: 0px 3px 6px rgba(0, 0, 0, 0.1);
  background-color: ${C.White};
  margin-bottom: 100px;
  border: 0.25px solid ${C.Border};
  z-index: 1;
  position: absolute;
  top: 25px;
  right: 0;
`;

export const MoreContainer = styled.div`
  position: relative;
`;

export const RightItemsContainer = styled(flexibleRowDiv)`
  position: absolute;
`;

export const DropdownItemContainer = styled(flexibleRowDiv)`
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 2.25rem;
  border-bottom: 0.25px solid ${C.Border};
  &:hover {
    cursor: pointer;
    background-color: ${C.IgnitusBlue};
    .title {
      color: ${C.HoverTextColor};
    }
  }
`;

export const DropdownItemText = styled(T.Paragraph)`
  color: ${C.IgnitusBlue};
  font-family: ${F.PrimaryFamily};
  text-align: center
  font-weight: ${F.Normal};
  font-size: ${F.SM};
`;
