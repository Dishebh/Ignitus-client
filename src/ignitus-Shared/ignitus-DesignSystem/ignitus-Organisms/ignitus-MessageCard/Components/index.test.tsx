import React from 'react';
import { mount, render } from 'enzyme';
import emotionSerializer, { matchers } from 'jest-emotion';
import { MessageCard } from './index';

import { Props } from '../types';
import * as S from '../styles';

expect.addSnapshotSerializer(emotionSerializer);
expect.extend(matchers);

const props: Props = {
  name: 'Donald Kunkel',
  position: 'Professor',
  numberMessages: '8 Messages',
  dateSent: 'Thu, 8:23AM',
  avatar: 'https://storage.googleapis.com/ignitus_assets/ig-avatars/eugene.png',
  online: true,
};

it('should rendered', () => {
  const tree = render(<MessageCard {...props} />);
  expect(tree).toMatchSnapshot();
});

describe('<MessageCard />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<MessageCard {...props} />);
  });

  it('should render ImageContainer', () => {
    expect(wrapper.find(S.ImageContainer).length).toBe(1);
    expect(wrapper.find('svg').length).toBe(1);
    expect(wrapper.find('img').length).toBe(1);
    expect(wrapper.find('img').prop('src')).toBe(props.avatar);
    expect(wrapper.find('circle').length).toBe(1);
  });

  it('should render MessageSubTitle', () => {
    expect(wrapper.find(S.MessageSubTitle).length).toBe(1);
    expect(wrapper.find(S.Sender).text()).toBe(props.name);
    expect(wrapper.find(S.SenderPosition).text()).toBe(props.position);
  });

  it('should render MessageDateTitle', () => {
    expect(wrapper.find(S.MessageDateTitle).length).toBe(1);
    expect(
      wrapper
        .find(S.Details)
        .at(0)
        .text(),
    ).toBe(props.numberMessages);
    expect(
      wrapper
        .find(S.Details)
        .at(1)
        .text(),
    ).toBe(props.dateSent);
  });
});
