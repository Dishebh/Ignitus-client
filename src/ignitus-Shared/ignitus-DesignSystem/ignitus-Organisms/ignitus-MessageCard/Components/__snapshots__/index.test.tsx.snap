// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendered 1`] = `
.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 1rem;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.05);
  background-color: #ffffff;
  box-sizing: border-box;
  padding: 1rem;
  width: 100%;
  margin-bottom: 1rem;
}

.emotion-0 {
  position: relative;
  padding: 0.5rem;
}

.emotion-0 img {
  border-radius: 50%;
  width: 4rem;
  max-width: 100%;
  max-height: 100%;
}

.emotion-0 svg {
  position: absolute;
  bottom: 0;
  right: 0;
  stroke: #ffffff;
  fill: #219653;
  height: 2rem;
  width: 2rem;
}

.emotion-1 {
  font-size: 16px;
  font-weight: 400;
  color: #2D2D2D;
  margin: 0;
  margin-left: 1rem;
}

.emotion-2 {
  font-size: 16px;
  font-weight: 400;
  color: #2D2D2D;
  margin: 0;
  font-weight: 700;
  font-size: 18px;
  color: #000066;
  margin-bottom: 0.5rem;
}

.emotion-3 {
  font-size: 16px;
  font-weight: 400;
  color: #2D2D2D;
  margin: 0;
  font-weight: 400;
  font-size: 16px;
  color: #000066;
  margin-bottom: 0.5rem;
}

.emotion-6 {
  text-align: right;
  margin-left: auto;
}

.emotion-4 {
  font-size: 16px;
  font-weight: 400;
  color: #2D2D2D;
  margin: 0;
  font-family: Open Sans;
  font-weight: 400;
  font-size: 16px;
  color: #9C9C9C;
  margin-bottom: 0.5rem;
}

<div
  class="emotion-7"
>
  <div
    class="emotion-0"
  >
    <svg>
      <circle
        cx="50%"
        cy="50%"
        r="30%"
        stroke-width="10%"
      />
    </svg>
    <img
      alt="avatar"
      src="https://storage.googleapis.com/ignitus_assets/ig-avatars/eugene.png"
    />
  </div>
  <p
    class="emotion-1"
  />
  <p
    class="emotion-2"
  >
    Donald Kunkel
  </p>
  <p
    class="emotion-3"
  >
    Professor
  </p>
  <p />
  <div
    class="emotion-6"
  >
    <p
      class="emotion-4"
    >
      8 Messages
    </p>
    <p
      class="emotion-4"
    >
      Thu, 8:23AM
    </p>
  </div>
</div>
`;
